src/modules/users/__tests__/users.controller.spec.ts

```diff
@@ -52,161 +53,313 @@ describe('modules/users/UsersController', () => {
-  });
-
-  it('should remove user from auth0', async () => {
-    const request = {
-      user: { auth0Id: '1234' },
-    };
-    const params = { id: '1234' };
-
-    usersService.findByAuth0Id = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock(request.user.auth0Id), roles: [] }));
-    usersService.findById = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock(params.id), auth0Id: 'abcd' }));
-    usersService.remove = jest.fn(() => Promise.resolve({ ok: 1 }));
-    mentorsService.removeAllApplicationsByUserId = jest.fn(() => Promise.resolve());
-    auth0Service.getAdminAccessToken = jest.fn(() => Promise.resolve({ access_token: '159' }));
-    auth0Service.deleteUser = jest.fn(() => Promise.resolve());
-
-    await usersController.remove(request, params)
-
-    expect(auth0Service.getAdminAccessToken).toHaveBeenCalled();
-    expect(auth0Service.deleteUser).toHaveBeenCalled();
-    expect(auth0Service.deleteUser).toHaveBeenCalledWith('159', 'abcd');
+  
+  describe('currentUser', () => {
+    it('should return the current user', async () => {
+      const request = { user: { auth0Id: '123' } };
+      const data: User = <User>({ _id: 123, name: 'Crysfel Villa'});
+      const response = { success: true, data };
+      usersService.findByAuth0Id = jest.fn(() => Promise.resolve(data));
+
+      expect(await usersController.currentUser(request)).toEqual(response);
+    });
+
+    it('should create a new user', async () => {
+      const request = { user: { auth0Id: '123' } };
+      const data: User = <User>({ _id: '123', name: 'Crysfel Villa' });
+      const response = { success: true, data };
+
+      usersService.findByAuth0Id = jest.fn(() => Promise.resolve(undefined));
+      usersService.findByEmail = jest.fn(() => Promise.resolve(undefined));
+      usersService.create = jest.fn(() => Promise.resolve(data));
+      emailService.send = jest.fn();
+      auth0Service.getAdminAccessToken = jest.fn(() => Promise.resolve({ access_token: 'abc' }));
+      auth0Service.getUserProfile = jest.fn(() => Promise.resolve({ _id: '123' }));
+
+      expect(await usersController.currentUser(request)).toEqual(response);
+      expect(usersService.create).toHaveBeenCalled();
+      expect(emailService.send).toHaveBeenCalled();
+    });
+    
+    it('should link an existing mentor', async () => {
+      const request = { user: { auth0Id: '123' } };
+      const data: User = <User>({ _id: '123', name: 'Crysfel Villa' });
+      const response = { success: true, data };
+
+      usersService.findByAuth0Id = jest.fn(() => Promise.resolve(undefined));
+      usersService.findByEmail = jest.fn(() => Promise.resolve(data));
+      usersService.update = jest.fn(() => Promise.resolve());
+      emailService.send = jest.fn();
+      auth0Service.getAdminAccessToken = jest.fn(() => Promise.resolve({ access_token: 'abc' }));
+      auth0Service.getUserProfile = jest.fn(() => Promise.resolve({ _id: '123' }));
```

Comment:
Given how often I see these mocks and stubs repeated in this spec alone, it may be worth thinking through an approach for Fixtures or [ObjectMothers](https://www.martinfowler.com/bliki/ObjectMother.html)/Factories.  In a typical Arrange/Act/Assert testing methodology, we spend a lot of time in the `Arrange` phase.  Anything that can be done to speed that part up is typically worth it.

Not for this PR, necessarily, but something to start brewing on.